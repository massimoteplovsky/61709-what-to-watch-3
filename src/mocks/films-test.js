export const films = [
  {
    id: 1,
    title: `Terminator 2. Judgment day`,
    genre: `Action`,
    year: 1991,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: [],
    src: ``,
    runtime: 5,
    reviews: [
      {
        text: ``,
        author: ``,
        date: ``,
        rating: 7.5
      }
    ]
  },
  {
    id: 2,
    title: `1+1`,
    genre: `Drama`,
    year: 2012,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: [],
    src: ``,
    runtime: 4,
    reviews: [
      {
        text: ``,
        author: ``,
        date: ``,
        rating: 7.5
      }
    ]
  },
  {
    id: 3,
    title: `Joker`,
    genre: `Drama`,
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: [],
    src: ``,
    runtime: 4,
    reviews: [
      {
        text: ``,
        author: ``,
        date: ``,
        rating: 7.5
      }
    ]
  },
  {
    id: 4,
    title: `Once upon a time in Hollywood`,
    genre: `Drama`,
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: [],
    src: ``,
    runtime: 4,
    reviews: [
      {
        text: ``,
        author: ``,
        date: ``,
        rating: 7.5
      }
    ]
  },
  {
    id: 5,
    title: `Ford vs Ferrari`,
    genre: `Drama`,
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: [],
    src: ``,
    runtime: 7,
    reviews: [
      {
        text: ``,
        author: ``,
        date: ``,
        rating: 7.5
      }
    ]
  },
  {
    id: 6,
    title: `1917`,
    genre: `Drama`,
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: [],
    src: ``,
    runtime: 4,
    reviews: [
      {
        text: ``,
        author: ``,
        date: ``,
        rating: 7.5
      }
    ]
  },
  {
    id: 7,
    title: `Parasite`,
    genre: `Drama`,
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: [],
    src: ``,
    runtime: 4,
    reviews: [
      {
        text: ``,
        author: ``,
        date: ``,
        rating: 7.5
      }
    ]
  },
  {
    id: 8,
    title: `Irishman`,
    genre: `Drama`,
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: [],
    src: ``,
    runtime: 4,
    reviews: [
      {
        text: ``,
        author: ``,
        date: ``,
        rating: 7.5
      }
    ]
  }
];

export const film = {
  id: 1,
  title: `Terminator 2. Judgment day`,
  genre: `Action`,
  year: 1991,
  promoPoster: ``,
  poster: ``,
  cover: ``,
  description: ``,
  rating: 8.5,
  votes: 12000,
  director: ``,
  starring: [`Tim cook`],
  src: ``,
  runtime: 4,
  reviews: [
    {
      text: ``,
      author: ``,
      date: ``,
      rating: 7.5
    }
  ]
};

export const promoFilmInfo = {
  id: 1,
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  year: 2014,
  poster: ``,
  src: ``
};

