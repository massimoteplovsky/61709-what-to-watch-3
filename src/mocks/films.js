export default [
  {
    id: 1,
    title: `Terminator 2. Judgment day`,
    poster: `http://i4.ytimg.com/vi/HgV7-MJwUBw/mqdefault.jpg`
  },
  {
    id: 2,
    title: `1+1`,
    poster: `https://i.pinimg.com/736x/76/44/c0/7644c07844dd4cb28d2c289bfa6e0dec--the-intouchables-comedy-movies.jpg`
  },
  {
    id: 3,
    title: `Joker`,
    poster: `https://all.culture.ru/uploads/9b743d135092e110a8e55a3dc3ad5bc4_w375_h250.jpeg`
  },
  {
    id: 4,
    title: `Once upon a time in Hollywood`,
    poster: `https://avatars.mds.yandex.net/get-pdb/1945878/3281b62e-4539-4eb7-a26b-fa68f6572bea/s375`
  },
  {
    id: 5,
    title: `Ford vs Ferrari`,
    poster: `https://sun9-28.userapi.com/c206816/v206816832/1771c/3p_wj8sfMz8.jpg`
  },
  {
    id: 6,
    title: `1917`,
    poster: `https://sun3-13.userapi.com/c857724/v857724955/14e882/u5CFMvbvX3I.jpg`
  },
  {
    id: 7,
    title: `Parasite`,
    poster: `http://m.kino-teatr.ru/art/5395/pv_75950.jpg`
  },
  {
    id: 8,
    title: `Irishman`,
    poster: `https://sun9-31.userapi.com/c850608/v850608676/40eb4/ojB5App6exk.jpg`
  },
];
