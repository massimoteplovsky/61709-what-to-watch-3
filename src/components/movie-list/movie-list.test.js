import React from "react";
import renderer from "react-test-renderer";
import {films} from "../../mocks/films-test";
import MovieList from "./movie-list.jsx";
import {Router} from "react-router-dom";
import history from "../../history";

it(`<MovieList /> component renders correctly`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <MovieList
            films={films}
          />
        </Router>
        , {
          createNodeMock: () => {
            return {};
          }}).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`<MovieList /> component renders correctly with empty films array`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <MovieList
            films={films}
          />
        </Router>
        , {
          createNodeMock: () => {
            return {};
          }}).toJSON();
  expect(tree).toMatchSnapshot();
});
