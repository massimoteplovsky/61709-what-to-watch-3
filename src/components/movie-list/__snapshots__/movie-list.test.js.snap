// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieList /> component renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        height="175"
        muted={true}
        poster=""
        src=""
        style={
          Object {
            "objectFit": "cover",
          }
        }
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Terminator 2. Judgment day
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        height="175"
        muted={true}
        poster=""
        src=""
        style={
          Object {
            "objectFit": "cover",
          }
        }
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        1+1
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        height="175"
        muted={true}
        poster=""
        src=""
        style={
          Object {
            "objectFit": "cover",
          }
        }
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Joker
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        height="175"
        muted={true}
        poster=""
        src=""
        style={
          Object {
            "objectFit": "cover",
          }
        }
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Once upon a time in Hollywood
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        height="175"
        muted={true}
        poster=""
        src=""
        style={
          Object {
            "objectFit": "cover",
          }
        }
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Ford vs Ferrari
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        height="175"
        muted={true}
        poster=""
        src=""
        style={
          Object {
            "objectFit": "cover",
          }
        }
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        1917
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        height="175"
        muted={true}
        poster=""
        src=""
        style={
          Object {
            "objectFit": "cover",
          }
        }
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Parasite
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        height="175"
        muted={true}
        poster=""
        src=""
        style={
          Object {
            "objectFit": "cover",
          }
        }
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Irishman
      </a>
    </h3>
  </article>
</div>
`;

exports[`<MovieList /> component renders correctly with empty films array 1`] = `
<h2>
  No movies
</h2>
`;
